### PR TITLE
Add JEOS_MULTINET configuration option

### PR DESCRIPTION
**Pull request by aplanas**
* Configure all interfaces where DHCP answer

* Fix grep stderr output in is_raspberry that happens in non-rpi devices

### CODE REVIEW

**Comment by aplanas** [comment]
Note that is on top of the other branch. I will rebase accordingly if the other branch gets merged.

**Comment by Vogtinator** [comment]
Question is whether the pre-MULTINET behaviour was actually useful.

IMO we can just use this by default, I don't see any downside in enabling DHCP for all interfaces. Which interface is first is also not really guaranteed anyway.

**Comment by aplanas** [comment]
@Vogtinator I agree with your comment. I would drop this logic of IPADDRESS vs GATEWAYS, but maybe there were a reason for that.

**Comment by Vogtinator** [comment]
@aplanas: I'd say: Go for it. Then we also don't need a config option.

**Comment by aplanas** [comment]
@Vogtinator done.

File: files/usr/share/defaults/jeos-firstboot.conf [hunk 1]
```diff
@@ -4,3 +4,4 @@
 # JEOS_KEYTABLE='en'
 # JEOS_TIMEZONE='UTC'
 # JEOS_PASSWORD_ALREADY_SET=0
+# JEOS_MULTINET=0
```
Comment (Vogtinator) [inline files/usr/share/defaults/jeos-firstboot.conf]: Not needed anymore

File: files/usr/lib/jeos-firstboot [hunk 2]
```diff
@@ -352,16 +352,14 @@ for p in /sys/class/net/* ; do
...
 
-        test -n "$IPADDR"   && candidates+=("$d")  || continue # ok, track it
-        test -n "$GATEWAYS" && candidates=("$d")  || continue # just use it
-        break
+	# Collect the interfaces that provides IPADDR and GATEWAYS
+	[ -n "$IPADDR" -a -n "$GATEWAYS" ] && candidates+=("$d")
 done
 
 for d in ${candidates[@]} ; do
         rm -f "/etc/sysconfig/network/ifcfg-$d" || exit 1
```
Comment (Vogtinator) [inline files/usr/lib/jeos-firstboot]: Can be merged into the loop above now and `candidates` removed

**Comment by Vogtinator** [comment]
IMO `GATEWAYS` should also be ignored. Getting an IP address is good enough for DHCP activation.

File: files/usr/share/defaults/jeos-firstboot.conf (hunk 1 above)
Comment (aplanas) [reply files/usr/share/defaults/jeos-firstboot.conf]: Done

File: files/usr/lib/jeos-firstboot (hunk 2 above)
Comment (aplanas) [reply files/usr/lib/jeos-firstboot]: Sure. Done

**Comment by aplanas** [comment]
> IMO `GATEWAYS` should also be ignored. Getting an IP address is good enough for DHCP activation.

I agree. I will change that according to @gmoro comments.

**Comment by gmoro** [comment]
> Question is whether the pre-MULTINET behaviour was actually useful.
> 
> IMO we can just use this by default, I don't see any downside in enabling DHCP for all interfaces. Which interface is first is also not really guaranteed anyway.
> 
> @gmoro: Any reason this was done this IMO overcomplicated way? I'd rather just enable DHCP for all interfaces which get an IP.

This part was actually coded by the maintainer of wicked, and the idea was to use the first configurable network, of course we can rethink it, if we have multiple nets which one are we going to use for the default route? What exactly would be the use case for multiple networks to be configured in the firstboot?

**Comment by Vogtinator** [comment]
> This part was actually coded by the maintainer of wicked, and the idea was to use the first configurable network, of course we can rethink it, if we have multiple nets which one are we going to use for the default route?

Neither of those options are perfect, but potentially having conflicting options for a default route (which shouldn't happen in a well defined environment) might be better than having to set up DHCP for other interfaces manually in case the selection was wrong.

That depends on wicked's behaviour for the conflict situation though, do you have any more info about that?

**Comment by aplanas** [comment]
> > @gmoro: Any reason this was done this IMO overcomplicated way? I'd rather just enable DHCP for all interfaces which get an IP.
> 
> This part was actually coded by the maintainer of wicked, and the idea was to use the first configurable network, of course we can rethink it, if we have multiple nets which one are we going to use for the default route? 

Uhm, I wonder if this is not task for the DHCP administrator to decide, and not for firstboot.

> What exactly would be the use case for multiple networks to be configured in the firstboot?

The use case is simple: derivatives from JeOS can be used in scenarios were multiple interfaces are available. In cloud and kubernetes cloud-init or ignition will configure the interfaces, but in other environments firstboot will need to assign some same defaults.

File: files/usr/lib/jeos-firstboot [hunk 3]
```diff
@@ -352,16 +351,13 @@ for p in /sys/class/net/* ; do
...
-done
-
-for d in ${candidates[@]} ; do
-        rm -f "/etc/sysconfig/network/ifcfg-$d" || exit 1
-        printf "STARTMODE=auto\nBOOTPROTO=dhcp\n" \
-                > "/etc/sysconfig/network/ifcfg-$d"
-        break # one is enough
+	# Create a configuration file for each interface that provides
+	# IPADDR and GATEWAYS
+	if [ -n "$IPADDR" -a -n "$GATEWAYS" ]; then
```
Comment (Vogtinator) [inline files/usr/lib/jeos-firstboot]: Remove the check for `GATEWAYS` here, interfaces without routes might still be useful.

Comment (aplanas) [reply files/usr/lib/jeos-firstboot]: Done. Note that for cleaning I also removed the `unset` and the `eval` parts.

**Comment by Vogtinator** [comment]
LGTM, just needs to be rebased on master after the other PR gets merged

**Comment by aplanas** [comment]
Rebased on top of the other PR (that is now also rebased from the new master)

**Comment by gmoro** [comment]
> That depends on wicked's behaviour for the conflict situation though, do you have any more info about that?

I don't, but let's go with it as it is.

**Comment by Vogtinator** [comment]
Don't merge before #49 got merged and this PR rebased on top.

**Pull request merged**